src/mito2/src/worker/handle_rebuild_index.rs

```diff
@@ -0,0 +1,205 @@
+        let region_id = request.region_id;
+        let Some(region) = self.regions.get_region(region_id) else {
+            return;
+        };
+
+        let version = region.version();
+
+        let all_files: HashMap<FileId, FileHandle> = version
+            .ssts
+            .levels()
+            .iter()
+            .flat_map(|level| level.files.iter())
+            .filter(|(_, handle)| !handle.is_deleted())
+            .map(|(id, handle)| (*id, handle.clone()))
+            .collect();
+
+        let build_tasks = if request.file_metas.is_empty() {
+            all_files.values().cloned().collect::<Vec<_>>()
+        } else {
+            request
+                .file_metas
+                .iter()
+                .filter_map(|meta| all_files.get(&meta.file_id).cloned())
+                .collect::<Vec<_>>()
+        };
+
+        for file_handle in build_tasks {
+            let _ = self
+                .index_build_scheduler
+                .schedule_build(self.new_index_build_task(
+                    &region,
+                    file_handle,
+                    request.build_type.clone(),
+                    None,
+                ));
+        }
+    }
+
+    pub(crate) async fn handle_index_build_finished(
+        &mut self,
+        region_id: RegionId,
+        request: IndexBuildFinished,
+    ) {
+        let region = match self.regions.get_region(region_id) {
+            Some(region) => region,
+            None => {
+                warn!(
+                    "Region not found for index build finished, region_id: {}",
+                    region_id
+                );
+                return;
+            }
+        };
+        let version = region.version();
+
+        for file_meta in &request.edit.files_to_add {
+            let file_id = file_meta.file_id;
+
+            // Check if the file exists in the region version.
+            let found_in_version = version
+                .ssts
+                .levels()
+                .iter()
+                .flat_map(|level| level.files.iter())
+                .any(|(id, handle)| *id == file_id && !handle.is_deleted() && !handle.compacting());
+            if !found_in_version {
+                warn!(
+                    "File id {} not found in region version for index build, region: {}",
+                    file_id, region_id
+                );
+                return;
+            }
+
+            // Check if the file exists in the object store.
+            let path = location::sst_file_path(
+                region.access_layer.table_dir(),
+                RegionFileId::new(file_meta.region_id, file_id),
+                PathType::Bare,
+            );
+            region.access_layer.object_store().exists(&path).await.map_err(|e| {
+                warn!(e; "SST file not found for index build, region: {}, file_id: {}", region_id, file_meta.file_id);
+            }).ok();
```

Comment:
Question: Why does it say "OK" when the file is missing? Is it correct to proceed with updating the manifest?